File: gdx2d-desktop/src/hevs/gdx2d/demos/scrolling/objects/Coin.java
package hevs.gdx2d.demos.scrolling.objects;

import hevs.gdx2d.components.bitmaps.BitmapImage;
import hevs.gdx2d.lib.GdxGraphics;
import hevs.gdx2d.lib.interfaces.DrawableObject;

/**
 * An animated coin that scales with time
 * @author Pierre-Andre Mudry (mui) 
 */
public class Coin implements DrawableObject {

	int posx, posy;	
	float scale = 0.4f, direction = 0.01f;	
	final BitmapImage img = new BitmapImage("data/images/retro-coin.png");
	
	public Coin(int x, int y){
		posx = x;
		posy = y;
	}
	
	@Override
	public void draw(GdxGraphics g) {
		g.drawTransformedPicture(posx, posy, 0, scale, img);
		
		// If too small or too big, change direction of scaling
		if(scale < 0.2 || scale > 0.45)
			direction *= -1;
		
		scale += direction;
	}

}

File: gdx2d-demoDesktop/src/main/scala/ch/hevs/gdx2d/demos/scrolling/objects/Coin.scala
package ch.hevs.gdx2d.demos.scrolling.objects

import ch.hevs.gdx2d.components.bitmaps.BitmapImage
import ch.hevs.gdx2d.lib.GdxGraphics
import ch.hevs.gdx2d.lib.interfaces.DrawableObject

/** An animated coin that scales with time. */
class Coin(var posx: Int, var posy: Int) extends DrawableObject {
  private val img = new BitmapImage("images/retro-coin.png")
  private var scale: Float = 0.4f
  private var direction: Float = 0.01f

  override def draw(g: GdxGraphics): Unit = {
    g.drawTransformedPicture(posx.toFloat, posy.toFloat, 0f, scale, img)
    if (scale < 0.2f || scale > 0.45f) direction *= -1f
    scale += direction
  }
}
